lib/mapObjects/CGTownInstance.h

```diff
@@ -73,6 +73,7 @@ class DLL_LINKAGE CGTownInstance : public CGDwelling, public IShipyard, public I
 	std::vector<std::vector<SpellID> > spells; //spells[level] -> vector of spells, first will be available in guild
 	std::vector<CCastleEvent> events;
 	std::pair<si32, si32> bonusValue;//var to store town bonuses (rampart = resources from mystic pond, factory = save debts);
+	int lastSpellResearchDay;
```

Comment:
Would also be nice to have "spell research allowed in town" boolean property - part of hota map format. Check `CMapLoaderH3M::readTown`